extern/guided_diffusion/guided_diffusion/train_util.py

```diff
@@ -340,16 +344,18 @@ def save_checkpoint(rate, params):
                 with bf.BlobFile(bf.join(get_blob_logdir(), filename), "wb") as f:
                     th.save(state_dict, f)
 
-        save_checkpoint(0, self.mp_trainer.master_params)
-        for rate, params in zip(self.ema_rate, self.ema_params):
-            save_checkpoint(rate, params)
+        save_checkpoint(0, self.mp_trainer.master_params)  # Save model
+
+        if self.save_ema:
+            for rate, params in zip(self.ema_rate, self.ema_params):
+                save_checkpoint(rate, params)  # Save EMA params
 
-        if dist.get_rank() == 0:
+        if self.save_opt and dist.get_rank() == 0:
             with bf.BlobFile(
                 bf.join(get_blob_logdir(), f"opt{(self.step+self.resume_step):06d}.pt"),
                 "wb",
             ) as f:
-                th.save(self.opt.state_dict(), f)
+                th.save(self.opt.state_dict(), f)  # Save optimizer params
```

Comment:
one of those cases where if you feel the need to add a comment, you should extract this one line to a function named `save_ema_params` and `save_opt_params`.
and then it will also be more readable.
your code should explain itself.

Reply:
I will not make an entire function for one function call that is fully self explainable if you take more than a second to read

Reply:
bad tal! i'll let this one go.